server/engine/src/main/java/org/eclipse/lsp/cobol/core/engine/processors/SectionNodeProcessorHelper.java

```diff
@@ -205,11 +205,16 @@ private static void adjustVariableNodeInsideCopyNode(CopyNode copyNode, List<Nod
   }
 
     private static boolean isVariableNodeAlreadyPresentIn(CopyNode copyNode, VariableDefinitionNode variable) {
-        return copyNode.getChildren().stream()
-                .filter(VariableDefinitionNode.class::isInstance)
-                .map(VariableDefinitionNode.class::cast)
-                .anyMatch(n -> n.getLocality().equals(variable.getLocality())
-                        && n.getVariableName().equals(variable.getVariableName()));
+      for (Node node : copyNode.getChildren()) {
+        if (node instanceof VariableDefinitionNode) {
+          VariableDefinitionNode varNode = (VariableDefinitionNode) node;
+          if (varNode.getLocality().equals(variable.getLocality())
+                  && varNode.getVariableName().equals(variable.getVariableName())) {
+            return true;
+          }
```

Comment:
```suggestion
          return varNode.getLocality().equals(variable.getLocality())
                  && varNode.getVariableName().equals(variable.getVariableName());
```

Reply:
Wouldn't it be check only for the first VariableDefinitionNode?

Reply:
sorry, missed the loop somehow...